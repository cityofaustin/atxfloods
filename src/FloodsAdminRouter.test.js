import React from 'react';
import ReactDOM from 'react-dom';
import { Route, MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import FloodsAdminRoutes from './FloodsAdminRoutes';
import Public from './Public';
import auth from './services/awsAuth';

jest.mock('./services/awsAuth', () => {
  return {
    isAuthenticated: jest.fn()
  }
});

it('renders the root page correctly when the user is logged out', () => {
  console.log(auth.isAuthenticated);
  auth.isAuthenticated.mockReturnValue(false);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders the root page correctly when the user is logged in', () => {
  auth.isAuthenticated.mockReturnValue(true);

  const tree = renderer.create(
    <MemoryRouter initialEntries={[ '/' ]}>
      <FloodsAdminRoutes />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
