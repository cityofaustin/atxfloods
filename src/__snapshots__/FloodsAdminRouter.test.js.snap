exports[`test renders the root page correctly when the user is logged in 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
</div>
`;

exports[`test renders the root page correctly when the user is logged out 1`] = `
<div>
  <p>
    You are not logged in.
  </p>
  <ul>
    <li>
      <a
        href="/public"
        onClick={[Function]}>
        Public Page
      </a>
    </li>
    <li>
      <a
        href="/protected"
        onClick={[Function]}>
        Protected Page
      </a>
    </li>
    <li>
      <a
        href="/createuser"
        onClick={[Function]}>
        Create User
      </a>
    </li>
  </ul>
</div>
`;
